src/test/java/com/commercetools/sync/services/ProductServiceTest.java

```diff
@@ -0,0 +1,96 @@
+package com.commercetools.sync.services;
+
+import com.commercetools.sync.products.ProductTestUtils;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.products.Product;
+import io.sphere.sdk.products.ProductDraft;
+import io.sphere.sdk.products.commands.ProductUpdateCommand;
+import io.sphere.sdk.products.commands.updateactions.ChangeName;
+import io.sphere.sdk.products.commands.updateactions.Publish;
+import io.sphere.sdk.queries.PagedQueryResult;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Collections.singletonList;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class ProductServiceTest {
+
+    private ProductService service;
+    private SphereClient ctpClient;
+
+    @Before
+    public void setUp() {
+        ctpClient = spy(SphereClient.class);
+        service = ProductService.of(ctpClient);
+    }
+
+    @Test
+    public void create() {
+        when(ctpClient.execute(any())).thenReturn(completedFuture(mock(Product.class)));
+
+        Product product = service.create(mock(ProductDraft.class)).toCompletableFuture().join();
+
+        assertThat(product).isNotNull();
```

Comment:
My point is just current Unit test implementation doesn't have much sense:
  
  1. You mock _any_ CTP client request, when it likely makes sense to mock by type (e.g., if you call create product method - mock CreateProductCommand call only.
  2. You verify only null validation of the result, when it makes sense to validate the service returns some certain data (in my example above it verifies that `service.create()` returns the value i exactly mock.

Same for `update`, `publish`, revert` and other method.

BUT, my point is: this is a simple service, and it's better to cover it by integration test:
create a draft, save it in CTP, read from CTP, update, verify the value is updated.

In this case Unit-test is redundant, because it will be re-covered by integration test.